deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/RegisterToLongleafJobTest.java

```diff
@@ -0,0 +1,240 @@
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.CdrDeposit;
+import org.apache.commons.io.FileUtils;
+import org.apache.jena.query.Dataset;
+import org.apache.jena.rdf.model.*;
+import org.apache.jena.tdb.TDBFactory;
+import org.apache.jena.vocabulary.RDF;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Set;
+
+import static edu.unc.lib.dl.test.TestHelpers.setField;
+import static org.junit.Assert.assertEquals;
+
+public class RegisterToLongleafJobTest extends AbstractDepositJobTest {
+    @Rule
+    public final TemporaryFolder tmpFolder = new TemporaryFolder();
+
+    private final static String FILE_SCHEME = "file:";
+    private final static String LOC1_ID = "/some/path/loc1";
+    private final static String LOC2_ID = "/some/path/loc2";
+    private final static String LOC3_ID = "/some/path/loc3";
+    private final static String FILE_CONTENT1 = "Some content";
+    private final static String FILE_CONTENT2 = "Other stuff";
+    private final static String MD5 = "MD5 checksum";
+
+    private RegisterToLongleafJob job;
+    private PID depositPid;
+    private Bag depBag;
+    private Model model;
+    private Path storageLocPath;
+    private String outputPath;
+    private String longleafScript;
+
+    @Before
+    public void init() throws Exception {
+        tmpFolder.create();
+
+        Dataset dataset = TDBFactory.createDataset();
+
+        job = new RegisterToLongleafJob();
+        job.setJobUUID(jobUUID);
+        job.setDepositUUID(depositUUID);
+        job.setDepositDirectory(depositDir);
+        setField(job, "dataset", dataset);
+
+        outputPath = tmpFolder.newFile().getPath();
+        longleafScript = getLongleafScript(outputPath);
+        job.setLongleafBaseCommand(longleafScript);
+
+        job.init();
+
+        // Get a writeable model
+        model = job.getWritableModel();
+
+        depositPid = job.getDepositPID();
+        storageLocPath = tmpFolder.newFolder("storageLoc").toPath();
+        depBag = model.createBag(depositPid.getRepositoryPath());
+        depBag.addProperty(Cdr.storageLocation, LOC1_ID);
+    }
+
+    @Test
+    public void registerSingleFileToLongleaf() throws Exception {
+        Bag workBag = addContainerObject(depBag, Cdr.Work);
+        Resource fileResc = addFileObject(workBag, FILE_CONTENT1, true);
+        fileResc.addProperty(CdrDeposit.storageUri, FILE_SCHEME + LOC2_ID + "/original_file");
+        workBag.addProperty(Cdr.primaryObject, fileResc);
+
+        job.closeModel();
+
+        job.run();
+
+        String registrationArguments = "register -f " + LOC2_ID  + "/original_file --checksums 'md5:" + MD5 + "'\n";
+        String output = FileUtils.readFileToString(new File(outputPath));
```

Comment:
add a StandardCharsets.UTF-8 to each of these `readFileToString`, my IDE gets grumpy without it :)  
Also, just a suggestion, but if you hold onto the `File` object representation of the outputPath when its first created you could save a bit of effort on instantiating new file objects.